### ISSUE TITLE
A question for the transformer-xl model

### ISSUE THREAD

**Issue by whoisltd** [chenxingphh/text-classification-pytorch#1]
Hi @chenxingphh 
I don't see memory input of pre hidden state in your model. Why do you not use this? In the Transformer-xl paper, I see the author had used this.

![image](https://user-images.githubusercontent.com/41822445/173173228-283dff4d-2655-4fcb-a840-ff36a974e07b.png)

with the text-classification task, don't need this? can you explain this to me? , thanks

**Comment by chenxingphh** [comment]
The reason Transformer-XL uses the pre hidden state is that the input in machine translation datasets is usually long, for example, it can be a paragraph, so the context alignment is expanded by using the pre hidden state. In this case, the input lengths are all short, so it is not needed.

**Comment by chenxingphh** [comment]
> Hi @chenxingphh I don't see memory input of pre hidden state in your model. Why do you not use this? In the Transformer-xl paper, I see the author had used this.
> 
> ![image](https://user-images.githubusercontent.com/41822445/173173228-283dff4d-2655-4fcb-a840-ff36a974e07b.png)
> 
> with the text-classification task, don't need this? can you explain this to me? , thanks

The reason Transformer-XL uses the pre hidden state is that the input in machine translation datasets is usually long, for example, it can be a paragraph, so the context alignment is expanded by using the pre hidden state. In this case, the input lengths are all short, so it is not needed.

**Comment by whoisltd** [comment]
thanks for your answer 🙌

**Comment by whoisltd** [comment]
Hello again!!! @chenxingphh 
I see your relative position embedding in your transformer xl is different from the author code. And when if I use memory from pre-segment, the loss will not reduce. Can you explain this? I was getting into trouble when building this model with TensorFlow :( Thanks!!!

**Comment by chenxingphh** [comment]
> relative position embedding

I'm not familiar with TF. For the reason why Loss will not reduce, I think it is possible to check whether the learning rate is set too small, or whether the parameters of the model will be updated during training. For the implemented relative position embedding, I think you can combine the implementation code according to the feedforward calculation formula as follows (the core code is line 131 to line139 in model_transformer_xl.py). Hope this reply can help you.
![rel](https://user-images.githubusercontent.com/27141267/176643962-150f9bbc-2613-4ba3-aca9-1cd2ed7e5925.png)

**Comment by chenxingphh** [comment]
> Hello again!!! @chenxingphh I see your relative position embedding in your transformer xl is different from the author code. And when if I use memory from pre-segment, the loss will not reduce. Can you explain this? I was getting into trouble when building this model with TensorFlow :( Thanks!!!

I'm not familiar with TF. For the reason why loss will not reduce, I think it is possible to check whether the learning rate is set too small, or whether the parameters of the model will be updated during training.  For the implemented relative position embedding, I think you can combine the implementation code according to the feedforward calculation formula of relative position embedding as follows (the core code is line 131 to line139 in model_transformer_xl.py). Hope this reply can help you.
![rel](https://user-images.githubusercontent.com/27141267/176644871-e9001e4e-65e6-4d24-8d36-b43e00841ede.png)
